src/nbl/video/CJITIncludeLoader.cpp

```diff
@@ -17,37 +17,34 @@ auto CJITIncludeLoader::getInclude(const system::path& searchPath, const std::st
 
 std::string CJITIncludeLoader::collectDeviceCaps(const SPhysicalDeviceLimits& limits, const SPhysicalDeviceFeatures& features)
 {
-    return R"===(
-#ifndef _NBL_BUILTIN_HLSL_JIT_DEVICE_CAPABILITIES_INCLUDED_
-#define _NBL_BUILTIN_HLSL_JIT_DEVICE_CAPABILITIES_INCLUDED_
-
-#include <nbl/builtin/hlsl/device_capabilities_traits.hlsl>
-
-namespace nbl
-{
-namespace hlsl
-{
-namespace jit
-{
-    struct device_capabilities
-    {
-        NBL_CONSTEXPR_STATIC_INLINE bool shaderFloat64 = )===" + std::to_string(limits.shaderFloat64) + R"===(;
-        NBL_CONSTEXPR_STATIC_INLINE bool subgroupArithmetic = )===" + std::to_string(limits.shaderSubgroupArithmetic) + R"===(;
-        NBL_CONSTEXPR_STATIC_INLINE bool fragmentShaderPixelInterlock = )===" + std::to_string(features.fragmentShaderPixelInterlock) + R"===(;
-
-        NBL_CONSTEXPR_STATIC_INLINE uint16_t maxOptimallyResidentWorkgroupInvocations = )===" + std::to_string(limits.maxOptimallyResidentWorkgroupInvocations) + R"===(;
-        NBL_CONSTEXPR_STATIC_INLINE uint32_t maxResidentInvocations = )===" + std::to_string(limits.maxResidentInvocations) + R"===(;
-    };
-
-//TODO: when `device_capabilities_traits` is ready
-//typedef nbl::hlsl::device_capabilities_traits<device_capabilities> device_capabilities_traits;
-//for now just alias them
-typedef device_capabilities device_capabilities_traits;
-}
-}
-}
-
-#endif  // _NBL_BUILTIN_HLSL_JIT_DEVICE_CAPABILITIES_INCLUDED_
-    )===";
+    #include "nbl/video/device_capabilities_traits_jit_members.hlsl"
+
+    std::string start = R"===(
+        #ifndef _NBL_BUILTIN_HLSL_JIT_DEVICE_CAPABILITIES_INCLUDED_
+        #define _NBL_BUILTIN_HLSL_JIT_DEVICE_CAPABILITIES_INCLUDED_
+        
+        #include <nbl/builtin/hlsl/device_capabilities_traits.hlsl>
+        
+        namespace nbl
+        {
+        namespace hlsl
+        {
+        namespace jit
+        {
+            struct device_capabilities
+            {
+                )===";
+
+    std::string end = R"===(    };
+        
+        typedef nbl::hlsl::device_capabilities_traits<device_capabilities> device_capabilities_traits;
```

Comment:
you can use `using` instead of `typedef` now

Reply:
Done